pagopa/io-app
# feat: [IA-711] Refactored payment confirmation user interface

## Short description
This PR is going to refactor the payment checkout section following the Figma mockup.

🟢 The e2e tests successfully run on https://github.com/pagopa/io-app/pull/3840/commits/d83a52d7471c37e8508fbf139d6a69458d815f4c.

~⚠️ The DE translation is missing, I just added it in english as a temporary fallback.~

⚠️ The `editPaypalPspEnabled` hardcoded flag has been moved into `ConfirmPaymentMethodScreen.tsx`.

### Preview
https://user-images.githubusercontent.com/5963683/159262348-b9bcd63f-7d1a-480c-811d-2dd468e53778.mov

### Accessibility preview
https://user-images.githubusercontent.com/5963683/159263403-a6982497-08a0-49e1-8ac5-fbfc7876482b.mov

## List of changes proposed in this pull request
- Refactored the whole user interface for the checkout page

## How to test
The whole payment flow should be able to be executed successfully. While on the checkout page it should be possible to switch payment method and PSP (at the moment only *not* using PayPal).

⚠️ Be aware that a lot of changes are in [ts/screens/wallet/payment/ConfirmPaymentMethodScreen.tsx](https://github.com/pagopa/io-app/pull/3840/files#diff-184732bd98509ed167a5e5235249a230941bf7a02584f947488257bfe970e978) and all the diffs should be loaded manually in the code review page on GitHub by clicking on `Load diff`.
